core/src/main/java/edu/wpi/grip/core/operations/composite/ContoursReport.java

```diff
@@ -1,15 +1,26 @@
 package edu.wpi.grip.core.operations.composite;
 
-import static org.bytedeco.javacpp.opencv_core.*;
+import edu.wpi.grip.core.operations.networktables.NTPublishable;
+import edu.wpi.grip.core.operations.networktables.NTValue;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.bytedeco.javacpp.opencv_core.MatVector;
+import static org.bytedeco.javacpp.opencv_core.Rect;
+import static org.bytedeco.javacpp.opencv_imgproc.boundingRect;
+import static org.bytedeco.javacpp.opencv_imgproc.contourArea;
 
 /**
  * The output of {@link FindContoursOperation}.  This stores a list of contours (which is basically a list of points) in
  * OpenCV objects, as well as the width and height of the image that the contours are from, to give context to the
  * points.
  */
-public final class ContoursReport {
+public final class ContoursReport implements NTPublishable {
```

Comment:
This would be easier if this class was immutable. Then this object wouldn't have to have to have thread locks on operations that should be simple reads.